MetaCell/cfos-visualizer
# Feature/cfos 97

**Tests implemented**

- Social Context Map test
  - Load the page: This test loads the application's URL, waits for certain elements to appear on the page, and checks that the text content of a specific element matches the expected value. It also takes a screenshot of a canvas element and compares it to a stored snapshot image to check for visual differences.
  - Change Social Context Map: This test simulates user interaction with the application. It hovers over and clicks on certain elements to change the social context map. It then checks that the text content of a specific element has changed to reflect the new map.
  - Context Map Comparison: This test takes another screenshot of the canvas element after the social context map has been changed. It compares this screenshot to a stored snapshot image to check for visual differences.

- Control Panel test
  - Load the page: This test loads the application's URL, waits for certain elements to appear on the page, and checks that the text content of a specific element matches the expected value. It also takes a screenshot of a canvas element and compares it to a stored snapshot image to check for visual differences.
  - Control Panel: Hide: This test simulates user interaction with the application. It clicks on a button to hide the Atlas Map, then takes a screenshot of the canvas element and compares it to a stored snapshot image to check for visual differences.
  - Control Panel: Show: This test simulates user interaction with the application. It clicks on a button to show the Atlas Map, then takes a screenshot of the canvas element and compares it to a stored snapshot image to check for visual differences.
  - Control Panel: Download: This test simulates user interaction with the application. It clicks on a button to download the Atlas Map, then checks the download response.

- Intensity test - skipped due to the slider changes being developed atm @ddelpiano 
  - Load the page: This test loads the application's URL, waits for certain elements to appear on the page, and checks that the text content of a specific element matches the expected value. It also takes a screenshot of a canvas element and compares it to a stored snapshot image to check for visual differences.
  - Change Global Intensity: This test simulates user interaction with the application. It moves a slider to change the global intensity of the application, then checks that the slider's position has been updated correctly.
  - Reset Global Intensity: This test simulates user interaction with the application. It clicks a button to reset the global intensity to its default value, then checks that the slider's position has been reset correctly.
  - Change individual Intensity: This test simulates user interaction with the application. It moves a different slider to change the individual intensity of the application, then checks that the slider's position has been updated correctly.
  - Reset individual Intensity: This test simulates user interaction with the application. It clicks a button to reset the individual intensity to its default value, then checks that the slider's position has been reset correctly.


- Viewer test
  - Load the page: This test loads the application's URL, waits for certain elements to appear on the page, and checks that the text content of a specific element matches the expected value. It also takes a screenshot of a canvas element and compares it to a stored snapshot image to check for visual differences.
  - Move to Previous Slices: This test simulates user interaction with the application. It clicks a button to move to the previous slices of the viewer, then takes a screenshot of the canvas element and compares it to a stored snapshot image to check for visual differences.
  - Center Stack: This test simulates user interaction with the application. It clicks a button to center the stack in the viewer, then takes a screenshot of the canvas element and compares it to a stored snapshot image to check for visual differences.
  - Move to Next Slices: This test simulates user interaction with the application. It clicks a button to move to the next slices of the viewer, then takes a screenshot of the canvas element and compares it to a stored snapshot image to check for visual differences.
  - Toggle Wireframe: This test simulates user interaction with the application. It clicks a button to switch to wireframe mode in the viewer, then takes a screenshot of the canvas element and compares it to a stored snapshot image to check for visual differences.


**Integration**

Tests are integrated into the dev pipeline, you can see a complete run with the tests included - https://g.codefresh.io/build/65fadfde6858b28d8d70af1a?step=e2e_tests&tab=output&logs=terminal

Some notes:
- Tests are running after the deploy step so they are testing what is being deployed
- Tests use the DOMAIN environmental variable as a first option for test url, if not available resorts to the default hard coded cfos.dev.metacell.us
